### PR TITLE
Make backend views use postgres models

### PR DESCRIPTION
**Pull request by gcapalbo**
Do not review until https://github.com/dimagi/commcare-hq/pull/10026 is merged.  Once that happens I'll merge with master and the diff will become more manageable.

### CODE REVIEW

**Comment by dimagimon** [comment]
There are 8 errors:

* corehq/apps/sms/forms.py, line 23 - F401 'get_available_backends' imported but unused
* corehq/apps/sms/models.py, line 1641 - E303 too many blank lines (2)
* corehq/apps/sms/views.py, line 13 - F401 'Q' imported but unused
* corehq/apps/sms/views.py, line 997 - F821 undefined name 'BadSMSConfigException'
* corehq/apps/sms/views.py, line 1324 - F821 undefined name 'BadSMSConfigException'
* corehq/messaging/smsbackends/grapevine/models.py, line 132 - F841 local variable 'resp' is assigned to but never used
* corehq/messaging/smsbackends/http/models.py, line 213 - F841 local variable 'response' is assigned to but never used
* corehq/messaging/smsbackends/megamobile/models.py, line 116 - F841 local variable 'response' is assigned to but never used

**Comment by dimagimon** [comment]
There are 9 errors:

* corehq/apps/sms/forms.py, line 23 - F401 'get_available_backends' imported but unused
* corehq/apps/sms/models.py, line 1641 - E303 too many blank lines (2)
* corehq/apps/sms/views.py, line 14 - F401 'Q' imported but unused
* corehq/apps/sms/views.py, line 548 - E303 too many blank lines (3)
* corehq/apps/sms/views.py, line 1021 - F821 undefined name 'BadSMSConfigException'
* corehq/apps/sms/views.py, line 1348 - F821 undefined name 'BadSMSConfigException'
* corehq/messaging/smsbackends/grapevine/models.py, line 132 - F841 local variable 'resp' is assigned to but never used
* corehq/messaging/smsbackends/http/models.py, line 213 - F841 local variable 'response' is assigned to but never used
* corehq/messaging/smsbackends/megamobile/models.py, line 116 - F841 local variable 'response' is assigned to but never used

**Comment by dimagimon** [comment]
There are 9 errors:

* corehq/apps/sms/forms.py, line 23 - F401 'get_available_backends' imported but unused
* corehq/apps/sms/models.py, line 1641 - E303 too many blank lines (2)
* corehq/apps/sms/views.py, line 14 - F401 'Q' imported but unused
* corehq/apps/sms/views.py, line 548 - E303 too many blank lines (3)
* corehq/apps/sms/views.py, line 1021 - F821 undefined name 'BadSMSConfigException'
* corehq/apps/sms/views.py, line 1348 - F821 undefined name 'BadSMSConfigException'
* corehq/messaging/smsbackends/grapevine/models.py, line 132 - F841 local variable 'resp' is assigned to but never used
* corehq/messaging/smsbackends/http/models.py, line 213 - F841 local variable 'response' is assigned to but never used
* corehq/messaging/smsbackends/megamobile/models.py, line 116 - F841 local variable 'response' is assigned to but never used

File: corehq/messaging/smsbackends/grapevine/models.py [hunk 1]
```diff
@@ -98,6 +94,43 @@ def get_available_extra_fields(cls):
...
+            affiliate_code=escape(config.affiliate_code),
+            auth_code=escape(config.authentication_code),
+            message=escape(text),
+            msisdn=escape(phone_number)
+        )
+
+        url = 'http://www.gvi.bms9.vine.co.za/httpInputhandler/ApplinkUpload'
+        req = urllib2.Request(url, data)
+        response = urllib2.urlopen(req, timeout=settings.SMS_GATEWAY_TIMEOUT)
+        resp = response.read()
```
Comment (dimagimon) [inline corehq/messaging/smsbackends/grapevine/models.py]: F841 local variable 'resp' is assigned to but never used

File: corehq/messaging/smsbackends/http/models.py [hunk 2]
```diff
@@ -175,3 +171,47 @@ def get_available_extra_fields(cls):
...
+        params[config.message_param] = text
+        params[config.number_param] = phone_number
+
+        url_params = urlencode(params)
+        try:
+            if config.method == "GET":
+                response = urlopen("%s?%s" % (config.url, url_params),
+                    timeout=settings.SMS_GATEWAY_TIMEOUT).read()
+            else:
+                response = urlopen(config.url, url_params,
```
Comment (dimagimon) [inline corehq/messaging/smsbackends/http/models.py]: F841 local variable 'response' is assigned to but never used

File: corehq/messaging/smsbackends/megamobile/models.py [hunk 3]
```diff
@@ -79,3 +75,42 @@ def get_available_extra_fields(cls):
...
+        config = self.config
+        params = urlencode({
+            "pid": pid,
+            "cel": phone_number,
+            "msg": text,
+            "src": config.source_identifier,
+        })
+        api_account_name = quote(config.api_account_name)
+        url = 'http://api.mymegamobile.com/%s?%s' % (api_account_name, params)
+        response = urlopen(url, timeout=settings.SMS_GATEWAY_TIMEOUT).read()
```
Comment (dimagimon) [inline corehq/messaging/smsbackends/megamobile/models.py]: F841 local variable 'response' is assigned to but never used

**Comment by gcapalbo** [comment]
@esoergel once this gets merged i'll do another pr to remove the couch gateway models and one more to remove the gateway couch views

**Comment by esoergel** [comment]
@gcapalbo I see you wrote
> Do not review until #10026 is merged. Once that happens I'll merge with master and the diff will become more manageable.

Has that happened already?

**Comment by gcapalbo** [comment]
@esoergel yes I merged with master and this is good to go

**Comment by esoergel** [comment]
Okay, I'll review first thing after lunch (sorry for the long delay)

**Comment by esoergel** [comment]
Merging before build because last commit is trivial and the previous one passed.
Also, mad props on the `+1,002 −2,832`!

**Comment by gcapalbo** [comment]
awesome. will merge master into https://github.com/dimagi/commcare-hq/pull/10100 now to make the diff on that one easier

**Comment by czue** [comment]
to the :moon: !

**Comment by dannyroberts** [comment]
Awesome work! Excited to see the effect this has on site-wide performance!

**Pull request merged**